src/main/java/com/denizenscript/depenizen/bukkit/events/jobs/JobsJobsExpGainScriptEvent.java

```diff
@@ -0,0 +1,125 @@
+package com.denizenscript.depenizen.bukkit.events.jobs;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.PlayerTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.events.ScriptEvent;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import com.denizenscript.depenizen.bukkit.objects.jobs.JobsJobTag;
+import com.gamingmesh.jobs.Jobs;
+import com.gamingmesh.jobs.api.JobsExpGainEvent;
+import com.gamingmesh.jobs.container.ActionType;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class JobsJobsExpGainScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // jobs player earns exp for <'job'> (action <'action'>)
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a player performs an action that would cause them to earn Jobs exp for a certain job
+    //
+    // @Context
+    // <context.job> Returns the job that the player is gaining exp for
+    // <context.experience> Returns the amount of exp the player will earn
+    // <context.action> Returns the name of the action being paid for, which can be: Break, StripLogs, TNTBreak, Place, Kill, MMKill, Fish, Craft, VTrade, Smelt, Brew, Enchant, Repair, Breed, Tame, Dye, Shear, Milk, Explore, Eat, custom-kill, Collect, Bake.
+    //
+    // @Determine
+    // "EXP:" + ElementTag(Decimal) to change the amount of Jobs exp this action should provide
+    //
+    // @Plugin Depenizen, Jobs
+    //
+    // @Player Always.
+    //
+    // @Group Depenizen
+    //
+    // -->
+
+    public JobsJobsExpGainScriptEvent() {
+        instance = this;
+        registerCouldMatcher("jobs player earns exp for <'job'> (action <'action'>)");
+    }
+
+    public static JobsJobsExpGainScriptEvent instance;
+    public JobsExpGainEvent event;
+    public JobsJobTag job;
+
+    @Override
+    public boolean couldMatch(ScriptPath path) {
+        if (!super.couldMatch(path)) {
+            return false;
+        }
+        if (!path.eventArgLowerAt(5).equals("job")
+                && Jobs.getJob(path.eventArgAt(5)) == null) {
+            return false;
+        }
+        if (!path.eventArgAt(7).isEmpty() && !path.eventArgLowerAt(7).equals("action")
+                && ActionType.getByName(path.eventArgAt(7)) == null) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean matches(ScriptEvent.ScriptPath path) {
+        if (!path.eventArgLowerAt(5).equals("job")
+                && !job.getJob().getName().equalsIgnoreCase(path.eventArgAt(5))) {
```

Comment:
Should use `runGenericCheck` rather than equalsIgnoreCase, to allow for matchers